rocketpy/motors/TankGeometry.py

```diff
@@ -0,0 +1,376 @@
+__author__ = "Pedro Henrique Marinho Bressan, Mateus Stano Junqueira"
+__copyright__ = "Copyright 20XX, RocketPy Team"
+__license__ = "MIT"
+
+import numpy as np
+
+from rocketpy.Function import Function, PiecewiseFunction, funcify_method
+
+try:
+    from functools import cache
+except ImportError:
+    from functools import lru_cache
+
+    cache = lru_cache(maxsize=None)
+
+try:
+    from functools import cached_property
+except ImportError:
+    from rocketpy.tools import cached_property
+
+
+class TankGeometry:
+    """Class to define the geometry of a tank. It is used to calculate the
+    geometrical properties such as volume, area and radius. The tank is
+    axi-symmetric, and its geometry is defined by a set of Functions that are
+    used to calculate the radius as a function of height.
+    """
+
+    def __init__(self, geometry_dict=dict()):
+        """Initialize TankGeometry class.
+
+        Parameters
+        ----------
+        geometry_dict : dict, optional
+            Dictionary containing the geometry of the tank. The geometry is
+            calculated by a PiecewiseFunction. Hence, the dict keys are disjoint
+            tuples containing the lower and upper bounds of the domain of the
+            corresponding Function, while the values correspond to the radius
+            function from a axis of symmetry.
+        """
+        self.geometry = geometry_dict
+        return None
+
+    @property
+    def geometry(self):
+        """
+        The dictionary containing the geometry of the tank.
+
+        Returns
+        -------
+        dict
+            Dictionary containing the geometry of the tank.
+        """
+        return self._geometry
+
+    @geometry.setter
+    def geometry(self, geometry_dict):
+        """
+        Sets the geometry of the tank.
+
+        Parameters
+        ----------
+        geometry_dict : dict
+            Dictionary containing the geometry of the tank.
+        """
+        self._geometry = dict()
+        for domain, function in geometry_dict.items():
+            self.add_geometry(domain, function)
+
+    @funcify_method("Height (m)", "radius (m)", extrapolation="zero")
+    def radius(self):
+        """
+        The radius of the tank as a function of height.
+
+        Returns
+        -------
+        rocketpy.Function
+            Piecewise defined Function of tank radius.
+        """
+        return self.radius
+
+    @cached_property
+    def average_radius(self):
+        """
+        The average radius of the tank.
+
+        Returns
+        -------
+        float
+            Average radius of the tank.
+        """
+        return self.radius.average(self.bottom, self.top)
```

Comment:
mmmmmm I'm not sure if the result here is what you expect.

This seems to be dependent on the discretization used.
If you set more points to define the geometry near the caps, the average radius will be lower than what it really looks like.

Maybe we should integrate the transversal area and divide it by the height?

Reply:
Could you elaborate a bit more your concerns with this implementation. Just to make sure we are on the same page, the formula for average used is the `average` method of `Function` class.

This method integrates the function and divides it by the domain length.

Reply:
I was thinking you were simply taking the average of the values, not doing the integral.
You guys were more prepared than expected hahahah

Well done!